### ISSUE TITLE
Landscape mode crash

### ISSUE THREAD

**Issue by awkure** [Corsage/Masterani#2]
Sometimes when trying to load episode from landscape mode the program crashes with the following error log.

![image](https://user-images.githubusercontent.com/21037731/34282173-97d31850-e6d4-11e7-85ee-9c9f2f000a55.png)

Which means that all tarnsactions need to be splitted into relatively small ones.

Perhaps, I should've probably PR this 🤔

**Comment by Corsage** [comment]
Yeah I've been noticing that as well. Working on a rewrite, but fairly busy right now. Thanks for testing it :)
